feeadjuster/feeadjuster.py

```diff
@@ -13,15 +13,57 @@
 plugin.update_threshold = 0.05
 
 
+def get_ratio_soft(our_percentage):
+    """
+    Basic algorithm: lesser difference than default
+    """
+    our_percentage = min(1, max(0, our_percentage))
+    return 10**(0.5 - our_percentage)
+
+
 def get_ratio(our_percentage):
     """
     Basic algorithm: the farther we are from the optimal case, the more we
     bump/lower.
     """
+    our_percentage = min(1, max(0, our_percentage))
     return 50**(0.5 - our_percentage)
 
 
+def get_ratio_hard(our_percentage):
+    """
+    Return value is between 0 and 20: 0 -> 20; 0.5 -> 1; 1 -> 0
+    """
+    our_percentage = min(1, max(0, our_percentage))
+    return 100**(0.5 - our_percentage) * (1 - our_percentage) * 2
+
+
+def get_chan_fees(plugin: Plugin, scid: str):
+    channels = plugin.rpc.listchannels(scid)["channels"]
+    for ch in channels:
+        if ch["source"] == plugin.our_node_id:
+            return { "base_fee_millisatoshi": ch["base_fee_millisatoshi"],
+                     "fee_per_millionth": ch["fee_per_millionth"] }
+
+
+def maybe_setchannelfee(plugin: Plugin, scid: str, base = None, ppm = None):
+    base = plugin.adj_basefee if base is None else base
+    ppm = plugin.adj_ppmfee if ppm is None else ppm
+    fees = get_chan_fees(plugin, scid)
```

Comment:
nit: i find it cleaner for the caller to explicitly pass `plugin.adj_basefee` and `plugin.adj_ppmfee` as `base` and `ppm` than checking if it's None here.